### ISSUE TITLE
Compile error with Typescript 3

### ISSUE THREAD

**Issue by rzane** [ktsn/vuex-type-helper#7]
I'm using typescript 3.1.6 and I'm getting this compile error:

```
node_modules/vuex-type-helper/helpers.d.ts:20:99 - error TS2536: Type 'Map[K]' cannot be used to index type 'T'.

20   <Key extends keyof T, Map extends Record<string, Key>>(map: Map): { [K in keyof Map]: (payload: T[Map[K]]) => R }
                                                                                                     ~~~~~~~~~
```

**Comment by ktsn** [comment]
This seems a TypeScript bug. https://github.com/Microsoft/TypeScript/issues/28269

**Comment by weswigham** [comment]
I do not believe this report to be a bug in TS - the type as written is (at least slightly) unsafe. Consider the instantiation:
```
T = { 0: number }
Key = 0
Map = {[i: string]: 0} & {0: "not zero"}
(keyof Map is thus string | 0)
K = 0
```
`T[Map[K]]]` in this case would not exist, since `Map[0]` is `"not zero"`, and `T["not zero"]` doesn't exist.

I'm not sure if that's exactly why the assignment is forbidden at the lowest level in the compiler, but since I could come up with valid inputs for the type variables for which the indexing operation is invalid (as indicated by the error), the error would seem to be correct.

**Comment by ktsn** [comment]
Oh thank you for clarifying it. I seem to have jumped to the conclusion.

**Comment by ktsn** [comment]
Fixed on v1.2.1.
